compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -214,13 +219,15 @@ impl<'a> Parser<'a> {
                 "try adding parentheses to match on a tuple...",
```

Comment:
Ideally this would skip the `...` if the or-pattern suggestion is not there. Otherwise it leads to confusing errors like:

```
error: unexpected `,` in pattern
  --> $DIR/issue-48492-tuple-destructure-missing-parens.rs:84:14
   |
LL |     let women, men: (Vec<Genome>, Vec<Genome>) = genomes.iter().cloned()
   |         -----^---- help: try adding parentheses to match on a tuple...: `(women, men)`
```

which looks like there's something else coming. Or you could just skip the `...` in both cases for both the tuple and the or-pattern suggestion; I believe there's precedence in other errors.

Reply:
Would something like this be acceptable?

```
            let msg = "try adding parentheses to match on a tuple...";
            err.span_suggestion(
                seq_span,
                if or_suggestion { msg } else { msg.split("...").next().unwrap() },
                format!("({})", seq_snippet),
                Applicability::MachineApplicable,
            );

```

I couldn't find a const way of slicing or splitting (Or concatenating with a variable name, unlike the concat! macro, and without an external dependency like https://github.com/Vurich/const-concat)

Reply:
Maybe use `msg.trim_end_matches('.')` instead? Either seems acceptable.